dlt/extract/utils.py

```diff
@@ -119,6 +132,45 @@ def check_compat_transformer(name: str, f: AnyFun, sig: inspect.Signature) -> in
     return meta_arg
 
 
+def wrap_async_iterator(
+    gen: AsyncIterator[TDataItems],
+) -> Generator[Awaitable[TDataItems], None, None]:
+    """Wraps an async generator into a list of awaitables"""
+    exhausted = False
+    busy = False
```

Comment:
is it better to pass some object instance to `run` function which will contain the state the variables?
```py
    exhausted = False
    busy = False
```